### ISSUE TITLE
auto margin

### ISSUE THREAD

**Issue by golunovas** [happynear/AMSoftmax#24]
Could you give a short introduction to this https://github.com/happynear/AMSoftmax/tree/master/prototxt/auto, please?

**Comment by happynear** [comment]
https://arxiv.org/pdf/1801.05599v3.pdf  Section 4

However, I think this work is not mature enough to be published, so I overlapped it with the original version...

**Comment by golunovas** [comment]
Thank you. If you don't mind I will leave the issue opened in case I will have some more questions about that.

**Comment by golunovas** [comment]
@happynear I'm trying to understand the meaning of the equation 9 from the paper. How did you come up with that equation? 
Also, I'm wondering if it's possible to use a similar approach for arcface or a-softmax. Do you have any thoughts about that?

**Comment by happynear** [comment]
This is not a mature work. I collected various statistics on several trained models and found this statistic matched the best margin. This statistic is only suitable for am-softmax because of the LSE function.

**Comment by golunovas** [comment]
Ok, thank you!
